consensus/safety-rules/src/persistent_safety_storage.rs

```diff
@@ -217,4 +225,48 @@ mod tests {
         assert_eq!(safety_data.last_voted_round, 8);
         assert_eq!(safety_data.preferred_round, 1);
     }
+
+    #[test]
+    fn test_waypoint_counters() {
+        let consensus_private_key = ValidatorSigner::from_int(0).private_key().clone();
```

Comment:
Looks good! If you're able to make the waypoint and safety data updates consistent (as I mentioned above), could you update this test (or add another test) that also checks that safety data, too? 😄

Reply:
Changed the existing `test()` to `test_safety_data_counters()` and added additional assertions for safety data counters after initializing and updating.